### PR TITLE
Auto label PRs that may require extra checks 

### PR DESCRIPTION
**Pull request by celinval**
### Description of changes: 

Add a new workflow that automatically adds the label "Z-BenchCI" to PRs that touch compiler, toolchain, dependencies and driver files that invoke our dependencies.

I also removed the label from the toolchain update job since this workflow will add the label once the PR is created.

### Resolved issues:

#2606

### Related RFC:

Optional #ISSUE-NUMBER.

### Testing:

* How is this change tested? I manually tested in my fork: https://github.com/celinval/kani-dev/pull/12

* Is this a refactor change? No

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+#
+# Configuration for auto-labeling PRs
+#
+# Note that we enable dot, so "**" matches all files in a folder
+
+Z-BenchCI:
+  - kani-compiler/**
+  - rust-toolchain.toml
+  - kani-dependencies
+  - kani-driver/src/call-*
```
Comment (zhassan-aws) [inline .github/labeler.yml:12]: Why are `call-*` files included?

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  - Cargo.lock
+
```
Comment (zhassan-aws) [inline .github/labeler.yml:14]: I think we should add `cprover_bindings/**` as well.

Comment (zhassan-aws) [reply .github/labeler.yml:14]: Perhaps also `library/**`?

File: .github/labeler.yml (hunk 1 above)
Comment (celinval) [reply .github/labeler.yml:12]: Because they may change how we invoke our dependencies.

Comment (celinval) [reply .github/labeler.yml:12]: Like a new cbmc flag

File: .github/labeler.yml (hunk 2 above)
Comment (celinval) [reply .github/labeler.yml:14]: Sure. I thought about adding library, but I'm assuming most changes are new APIs or documentation. Anyway, we can manually add or remove the label if we feel like it's not needed

**Review by zhassan-aws — APPROVED** [review]

**Pull request merged**